tests/test_BlkDiagMatrix.py

```diff
@@ -335,3 +335,126 @@ def testBlkDiagMatrixArith(self):
             results.append(res)
 
         self.assertTrue(np.allclose(results[0], results[1].dense()))
+
+
+class IrrBlkDiagMatrixTestCase(TestCase):
```

Comment:
Some of these tests (like `apply`) generalize the ones for square matrices, so we can remove the corresponding tests there, right?

Reply:
We probably should not stop testing square arrays just because we can now admit rectangular ones too. So there would be two collections of cases either way... its just a matter of how to organize them.

I can put them (square and non-square) together, but I'd really rather not organize it that way. When any issues/bugs arise it will be much more useful to a developer to have the cases distinctly separate compared to only having the more complicated case. or having them all smashed together.  Generally, I think the intent of the tests is much clearer when separated.

I do appreciate that `apply` is copy pasta (and ugly), but it would be either way...

I might pass on this unless you really feel there is value to it.

Reply:
Ok that's fair.